### ISSUE TITLE
Support multiple workspace types per scheduler

### ISSUE THREAD

**Issue by d4l3k** [meta-pytorch/torchx#590]
## Description

Currently a scheduler is tied to a specific image type. Some schedulers such as LSF (and Slurm) support multiple different types of images. Ideally we could allow swapping in multiple different workspace backends to enable this in a more easy to use way.

## Detailed Proposal

1. Add some new syntax to the `--workspace` entrypoint to allow specifying different types of workspaces `--workspace=docker:.` or perhaps with a `--workspace-type docker`.

2. Change the Workspace interaction with the schedulers to allow for setting more than one available workspace type per scheduler.

```python
class FooScheduler(Scheduler):
    WORKSPACES: Iterable[Workspace] = (DockerWorkspace, DirWorkspace)
```

3. Update runner to be aware of the new workspace selectors

## Additional context/links

* Will depend on #589  since it removes workspace specific configs from runopts
* #588 - lsf scheduler PR

**Comment by kurman** [comment]
Will workspace still be synonymous with a 'patch' here?

**Comment by d4l3k** [comment]
Yeah -- would be. Just allows for different types of images/patching with the same scheduler
